<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Philip Ashton" />


<title>Genomic epidemiology practical (no command line version)</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>




<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 60px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h2 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h3 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h4 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h5 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h6 {
  padding-top: 65px;
  margin-top: -65px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Light phylogeny practical</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Genomic epidemiology practical (no command line version)</h1>
<h4 class="author"><em>Philip Ashton</em></h4>
<h4 class="date"><em>28 May, 2019</em></h4>

</div>


<div id="learning-objectives" class="section level1 unnumbered">
<h1>Learning objectives</h1>
<ol style="list-style-type: decimal">
<li><p>Be able to interpret SNP distance histograms.</p></li>
<li><p>Be able to interpret phylogenetic trees for epidemiological purposes.</p></li>
<li><p>Be familiar with various tree visualisation options.</p></li>
<li><p>Have a grasp on the relationship between trees and SNPs.</p></li>
</ol>
</div>
<div id="part-1" class="section level1 unnumbered">
<h1>Part 1</h1>
<p>Inject 1 - In the aftermath of a massive earthquake in Haiti, the poorest country in the Americas, there has been a dramatic and worrying increase in the number of cases of cholera, the disease caused by the bacterium <em>Vibrio cholera</em>. This is the first time in more than 100 years that there has been a cholera outbreak in Haiti. Urgent efforts are underway to identify the source of the outbreak.</p>
<p>Fortunately, in addition to field epidemiologists carrying out shoe-leather investigations in the affected areas, we have WGS of some of the isolates from the outbreak. You are the lead bioinformatician in charge of analysing this data. You need to process and interpret the data, and feedback to the public health doctors in charge of outbreak response to provide them with the fullest picture possible.</p>
<p>From using Google, or your microbiology knowledge about cholera:</p>
<p><strong>Question 1:</strong> Why might there be an increase in cholera cases following an event like an earthquake?</p>
<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here...">
</textarea>
<p><strong>Question 2:</strong> What is unusual about the genome of <em>V. cholera</em>?</p>
<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here...">
</textarea>
<p>Here is the pairwise SNP distance histogram for the genomes in our current collection:</p>
<div class="figure">
<img src="https://imgur.com/KDGGsue.png" alt="SNP pairwise-distance histogram" />
<p class="caption">SNP pairwise-distance histogram</p>
</div>
<p><strong>Question 3:</strong> What is the median pairwise SNP distance? How well do you think this describes the dataset?</p>
<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here...">
</textarea>
<p><strong>Question 4: </strong> What does the histogram tell you about the relatedness of the samples in your dataset?</p>
<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here...">
</textarea>
<p><strong>Question 5:</strong> Why are there 780 ‘samples’ in the output?</p>
<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here...">
</textarea>
<div class="alert alert-info">
<p>ALERT - READ ME</p>
<ul>
<li><p>At this point, you have to imagine that you’ve run an IQ-TREE command to ‘convert’ your sequence files into a phylogenetic tree.</p></li>
<li>In the interests of simplicity, I’ve done this for you and you can download the finished tree from below</li>
</ul>
</div>
<ul>
<li><p>Now, download the phylogenetic tree from <a href="https://www.dropbox.com/s/ft731tm7lmtkc7c/dataset1.aln.nwk?dl=0" target="_blank">here</a> , and a meta-data file with information on the geographic location from <a href="https://www.dropbox.com/s/xjxldv8ds0wai40/geog_loc_microreact.csv?dl=0" target="_blank">here</a> .</p></li>
<li><p>We are going to use Microreact to visualise the tree and associated metadata. Navigate to <a href="https://microreact.org/upload" target="_blank">here</a> and upload the <code>dataset1.aln.nwk</code> file and the <code>geog_loc_microreact.csv</code> file. The Microreact interface is hopefully quite intuitive, spend a few minutes becoming familiar with the various options and buttons. Microreact is an ideal way to investigate relationships between phylogeny and geography (aka phylogeographical relationships).</p></li>
</ul>
<p><strong>Question/task 6:</strong> Identify the Haiti outbreak genomes in the tree.</p>
<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here...">
</textarea>
<div class="toggle">
<button>
Hint
</button>
<p>They are <strong>most</strong> of the non-black tips.</p>
</div>
<p><br></p>
<ul>
<li>Experiment with the different tree visualisation setting. The different tree settings look like this:</li>
</ul>
<center>
<img src="images/tree_settings.png" />
</center>
<p><strong>Question 7:</strong> Are the isolates from Haiti a monophyletic group?</p>
<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here...">
</textarea>
<div class="toggle">
<button>
Hint
</button>
<p>Remember - a monophyletic group is a group which consists of all the descendents of a common ancestor.</p>
</div>
<p><br></p>
<p><strong>Question 8:</strong> Is there phylogeographical signal in the <em>V. cholerae</em> genomes from <strong>within</strong> Haiti? You will have to zoom in on Haiti to see this. I.e. are the isolates from different regions of Haiti genetically distinct from each other?</p>
<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here...">
</textarea>
<div class="toggle">
<button>
Hint
</button>
<p>You can think of this another way, looking at the data, if you were given another Haitian isolate with no geographical information, how confident would you be in your prediction of it’s geographic origin?</p>
</div>
<p><br></p>
<p><strong>Question 9:</strong> Where is the closest neighbour of the Haiti genomes from?</p>
<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here...">
</textarea>
<p><strong>Question 10:</strong> Write a short paragraph, summarising your results for the public health officials.</p>
<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here...">
</textarea>
</div>
<div id="part-2" class="section level1 unnumbered">
<h1>Part 2</h1>
<p>Inject 2 - Using the feedback we gave them on the likely origin of the <em>V. cholera</em> clade causing the outbreak, public health officials have traced the likely source of the cholera outbreak to the UN Stabilisation Mission in Haiti Camp. A contingent of Nepalese soldiers had recently arrived in camp, and cholera is endemic in Nepal. Sewage from the camp is discharged into the Artibonite River, where many people draw their drinking water from. This is an incredibly sensitive subject because, while identifying the true source is very important, if the source is mis-identified, it could lead to a major diplomatic incident.</p>
<p>Therefore, you have decided to search the public databases for all sequences from Nepal and Bangladesh and re-run the analysis, including these genomes. This will provide extra context for the Haitian outbreak and increase our certainty of what is happening.</p>
<div class="alert alert-info">
<p>ALERT - READ ME</p>
<ul>
<li><p>I’ve processed some additional genomic datasets and combined them with intial set you’ve been working with.</p></li>
<li>I then re-calculated a new phylogenetic tree from this larger dataset.</li>
</ul>
</div>
<p>You can download the tree from <a href="https://www.dropbox.com/s/ugni3gbt8uargbf/dataset2.aln.pretty.nwk?dl=0" target="_blank">here</a>, the annotation for figtree from <a href="https://www.dropbox.com/s/ylepng6vb7y71n5/annotation_figtree.txt?dl=0" target="_blank">here</a> and the annotation for phandango from <a href="https://www.dropbox.com/s/2rwkqf3di7x0l7u/annotation_phandango.csv?dl=0" target="_blank">here</a>.</p>
<ul>
<li><p>Open <code>dataset2.aln.pretty.nwk</code> with <code>FigTree</code>.</p></li>
<li><p>In <code>FigTree</code>, go to <code>File -&gt; Import annotations</code> in the menu bar, and navigate to the <code>annotation_figtree.txt</code> file you downloaded. Select this file and press open. Then in the main <code>FigTree</code> window, go to <code>Tip Labels</code>, and in the drop down <code>Display</code> menu, select <code>ANNOT</code>. You should now be able to see tip labels containing information about continent, country and year on the tree.</p></li>
</ul>
<p><strong>Question 1:</strong> <code>SRR308716</code> was the most closely related non-Haitian genome to the Haiti outbreak strains in <code>dataset1</code>. Is it still the most closely related genome?</p>
<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here...">
</textarea>
<p><strong>Question 2:</strong> Do all the genomes from Nepal form a monophyletic group? What does this tell you about cholera in Nepal?</p>
<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here...">
</textarea>
<p><strong>Question 3:</strong> What does the root of the tree represent?</p>
<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here...">
</textarea>
<ul>
<li>On the lefthand side of the <code>Figtree</code> interface, tick the box which says ‘Node labels’, click the arrow to enable you to select ‘label’ from the dropdown <code>Display</code> menu. This will place the bootstraps on the nodes of the tree.</li>
</ul>
<p><strong>Question 4:</strong> What is the impact of this extra context on your report to the public health doctors? Write a short paragraph reporting your new findings.</p>
<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here...">
</textarea>
<p><strong>Extra task:</strong> Use the <code>phandango</code> tool to visualise the meta-data in a more interesting way. Navigate to <a href="https://jameshadfield.github.io/phandango/#/" target="_blank">here</a> . And drag and drop your tree and the <code>annotation_phandango.csv</code> file onto the web page. Pressing <code>k</code> will bring up the key. Further instructions are <a href="https://github.com/jameshadfield/phandango/wiki" target="_blank">here</a> .</p>
<p><strong>Advanced question 1:</strong> Using the phylogenetic tree, estimate the SNP distance between <code>SRR308691</code> and <code>SRR308715</code>. A piece of paper might help you estimate the distance between the two tips. Remember you have to go from one tip, down to the MRCA, back to the other tip. Only ‘horizontal’ distance (if you are viewing the tree in rectangular format) counts. The other information you need to calculate this is the tree length, which is 0.029 (from IQTREE log) and the alignment length, which is 987 bp (calculated by <code>bioawk -c fastx '{ print $name, length($seq) }' &lt; dataset2.aln</code>).</p>
<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here...">
</textarea>
<div class="toggle">
<button>
Hint
</button>
<p>Comes from this formula: (<code>distance between nodes</code> / <code>total tree length</code>) * <code>alignment length</code></p>
</div>
<p><br></p>
<div id="notes" class="section level2 unnumbered">
<h2>Notes</h2>
<p>For more information on <em>V. cholera</em> genomics you can see <a href="https://www.ncbi.nlm.nih.gov/pubmed/29123068" target="_blank">here</a> . There are multiple papers on the genomics of the Haiti cholera outbreak, I leaned on <a href="https://mbio.asm.org/content/5/6/e01721-14" target="_blank">this</a> one to write this material.</p>
<script>
  $(".toggle").click(function() {
    $(this).toggleClass("open");
  });
</script>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
